### PR TITLE
Distributed Groups, __modified timestamp, and Downwards Synchronization

### PR DESCRIPTION
**Pull request by chrismerck**

### CODE REVIEW

File: common/info.yaml [hunk 1]
```diff
@@ -297,6 +313,295 @@ description: |
...
+      }
+
+      KSMJWCE12345:groups/0000000000000001 = {
+        "__modified": 1642788968,
+        "name": "Great Room Fans",
+        "types": ["CF"],
+        "locations": ["Great Room"],
+        "devices": ["1"]
+        "actions": ["TurnOn", "TurnOff", "TurnLightOn", "TurnLightOff"]
+      }
```
Comment (endy-s) [inline common/info.yaml:397]: I believe you meant ID 2. ID 1 is ZZ only.
```suggestion
  group with ID 2 is:

      ZZBL12345:groups/0000000000000002 = {
        "__modified": 1642788966,
        "name": "Great Room Fans",
        "types": ["CF"],
        "locations": ["Great Room"],
        "devices": ["aabbccdd", "11223344"],
        "actions": ["TurnOn", "TurnOff"]
      }

      KSMJWCE12345:groups/0000000000000002 = {
        "__modified": 1642788968,
        "name": "Great Room Fans",
        "types": ["CF"],
        "locations": ["Great Room"],
        "devices": ["1"]
        "actions": ["TurnOn", "TurnOff", "TurnLightOn", "TurnLightOff"]
      }
```

File: common/info.yaml [hunk 2]
```diff
@@ -297,6 +313,295 @@ description: |
...
+
+      {"power": 1}
+
+  If now one of the member devices, say the Smart by Bond fan, is turned off,
+  then the client would receive a state update for the device and group:
+
+      KSMJWCE12345:devices/1/state = {
+        "power": 0,
+        "light": 0
+      }
```
Comment (endy-s) [inline common/info.yaml:486]: Wrong device state:
```suggestion
      ZZBL12345:groups/0000000000000001/state = {
        "power": 1,
      }
```

File: common/info.yaml [hunk 3]
```diff
@@ -297,6 +313,295 @@ description: |
...
+
+      > POST http://TWCTBXX00001/v2/groups/0000000000000001/skeds {"action":"SetBrightness", "argument":75, "mark":"sunrise", "seconds":0}
+      < 201 OK {"_post_id": "0000000000000007"}
+
+      > POST http://TWCTBXX00002/v2/groups/0000000000000001/skeds {"_post_id": "0000000000000007", "action":"SetBrightness", "argument":75, "mark":"sunrise", "seconds":0}
+      < 201 OK {"_post_id": "0000000000000007"}
+
+      > POST http://TWCTBXX00003/v2/groups/0000000000000001/skeds {"_post_id": "0000000000000007", "action":"SetBrightness", "argument":75, "mark":"sunrise", "seconds":0}
+      < 201 OK {"_post_id": "0000000000000007"}
+
```
Comment (endy-s) [inline common/info.yaml:580]: May be good to add the parallelization info here too.
```suggestion

  The last two requests could be parallelized.

```

File: common/info.yaml [hunk 4]
```diff
@@ -297,6 +313,295 @@ description: |
...
+  In the event of one or more Bonds being offline, they will continue to execute
+  the sked until the Bond Home app syncs with them and removes the sked
+  which was supposed to be deleted. The Bond Home app knows that a sked
+  was deleted by checking the `__modified` timestamp on the group shard skeds
+  enumeration `groups/{}/skeds`. If a sked is present on an older enumeration
+  but absent on a newer one, it will be removed from the old enumeration.
+  Similarly, if a sked is present on the *newer* enumeration but absent on the
+  old one, then the Bond Home app with downwards-synchronize by POSTing the
+  missing sked.
```
Comment (endy-s) [inline common/info.yaml:603]:
```suggestion
  old one, then the Bond Home app adds it with downwards-synchronize 
  by POSTing the missing sked.
```

Comment (chrismerck) [reply common/info.yaml:603]: done tnx

**Comment by chrismerck** [comment]
I added a warning that the client must set the Request ID same when sending parallel actions to the shards. But that may be too much to ask of the integrators... unless we somehow simplify the requestor/request-id stuff in V3. -- Would be nice to have proper requestor info for 3rd-party integrators.

Maybe we don't need to ask for same request ID. Later we could add a different flag indicating that the client wants the Bond to act as a central controller, repeating / merging the requests and reporting back.... So, I'll remove that Request ID WARNING.

**Review by fredericojordan — APPROVED** [review]
Nice! 🎉 
Apart from the correction Endy pointed out, the rest is good to go!

**Review by marciogranzotto — APPROVED** [review]

**Pull request merged**